pkg/cache/multi_namespace_cache.go

```diff
@@ -185,7 +185,7 @@ func (c *multiNamespaceCache) WaitForCacheSync(ctx context.Context) bool {
 func (c *multiNamespaceCache) IndexField(ctx context.Context, obj client.Object, field string, extractValue client.IndexerFunc) error {
 	isNamespaced, err := objectutil.IsAPINamespaced(obj, c.Scheme, c.RESTMapper)
 	if err != nil {
-		return nil //nolint:nilerr
+		return nil //nolint:ignore
```

Comment:
@Vaxuite @alvaroaleman, it shows that what is not right here is we change the liners. 
However, to make C+R begin to use go 1.18, why should we not bump the modules?
I could not follow that.

Reply:
> I'm with @alvaroaleman here; I'm not sure if there's an immediate reason to update the go.mod's go version unless we do that on purpose to support go 1.18 features

usually, we bump the k8s version and go modules on deps before we upgrade Kubebuilder
Why?
- We want to ensure the same go version across the projects in the same ecosystem. Otherwise, it is hard for contributors to switch the projects. 
- We faced problems in the past based on incompatibilities, i.e. with go 1.17:

> For example, with some recent changes, we can't use k8s 1.23 with older versions of Go (For ex: https://github.com/operator-framework/operator-sdk/issues/5558#issuecomment-1047137754) - this is a vice versa case.

Then, I do not understand how controller-runtime is using go 1.18 when its modules are not bumped with 1.18?

Reply:
So when you specify the go version in the module file its a minimum version the module must run with. So with this module on go 1.17, if you run it with go 1.18 that will work fine and this does not really need changing.

Reply:
What made you think that this module needed to be bumped in the first place? @camilamacedo86

Reply:
The motivation here is not bump to allow: 
> By keeping it on 1.17 we allow consumers of CR to use Go 1.17+.
But we are using go 1.18 in the ci, see: https://kubernetes.slack.com/archives/C02MRBMN00Z/p1652178727902549?thread_ts=1652177106.531319&cid=C02MRBMN00Z

+1 this approach has my support either. 
and we should do the same in Kubebuilder.

Reply:
/hold